pkg/operator/ceph/cluster/cluster_external.go

```diff
@@ -112,6 +113,19 @@ func (c *ClusterController) configureExternalCephCluster(cluster *cluster) error
 		return errors.Wrap(err, "failed to create csi config map")
 	}
 
+	// update the msgr2 flag
+	for _, m := range cluster.ClusterInfo.Monitors {
+		// m.Endpoint=10.1.115.104:3300
+		monPort := util.GetPortFromEndpoint(m.Endpoint)
+		if monPort == client.Msgr2port {
+			externalConnections := &cephv1.ConnectionsSpec{
+				RequireMsgr2: true,
+			}
+			cluster.Spec.Network.Connections = externalConnections
+			break
```

Comment:
Let's also add a log statement indicating that msgr2 is required

Reply:
Updated

Reply:
How about if we also log a warning if all mons are not using the same port? This could be a new commit in this PR.

Reply:
in external cluster it is expected to have one port as v2 and other as v1, so we should not warn.
Thanks